test/extended/router/gatewayapicontroller.go

```diff
@@ -673,3 +760,164 @@ func isOKD(oc *exutil.CLI) (bool, error) {
+
+		if err != nil {
+			e2e.Logf("Failed to check %s, error: %v, retrying...", ossmSubscriptionName, err)
+			return false, nil
+		}
+
+		if unhealthy != "False" {
+			e2e.Logf("Wait CatalogSourcesUnhealthy status to be False, and got %s", unhealthy)
+			return false, nil
+		}
+
+		return true, nil
+	})
+	o.Expect(err).NotTo(o.HaveOccurred())
+}
+
+// used to wait an OSSM csv is automatically created successfully
+func pollWaitOssmCsvCreated(oc *exutil.CLI, operatorNamespace, csvName string) {
+	err := wait.Poll(3*time.Second, 300*time.Second, func() (bool, error) {
+		phase, err := oc.AsAdmin().WithoutNamespace().Run("get").Args("-n", operatorNamespace, "csv", csvName, `-o=jsonpath={.status.phase}`).Output()
+		if err != nil {
+			e2e.Logf("Failed to check %s, error: %v, retrying...", csvName, err)
+			return false, nil
+		}
+		if phase != "Succeeded" {
+			e2e.Logf(fmt.Sprintf("Wait for phase to be Succeeded, and got %s", phase))
+			return false, nil
+		}
+		return true, nil
+
+	})
+	o.Expect(err).NotTo(o.HaveOccurred())
+}
+
+// used to delete a deployment and wait for it is automatically recreated again
+func deleteDeploymentAndWaitAvailableAgain(oc *exutil.CLI, deploymentName, ns string) {
+	g.By(fmt.Sprintf("Try to delete the deployment %s in %s namespace", deploymentName, ns))
+	err := oc.AdminKubeClient().AppsV1().Deployments(ns).Delete(context.Background(), deploymentName, metav1.DeleteOptions{})
+	o.Expect(err).NotTo(o.HaveOccurred())
+
+	g.By(fmt.Sprintf("Wait until the deployment %s in %s namespace is recreated and returns back healthy", deploymentName, ns))
+	err = wait.Poll(3*time.Second, 300*time.Second, func() (bool, error) {
+		deployment, err := oc.AdminKubeClient().AppsV1().Deployments(ns).Get(context.Background(), deploymentName, metav1.GetOptions{})
+		if err != nil {
+			if errors.IsNotFound(err) {
+				return false, nil
+			}
+			e2e.Logf("Error getting %s deployment: %v, retrying", deploymentName, err)
+			return false, nil
+		}
+
+		readyReplicas := deployment.Status.ReadyReplicas
+		e2e.Logf("The ready replicas is %v", readyReplicas)
+		if readyReplicas != 1 {
+			e2e.Logf(`The deployment %s in %s namespace is not ready(AvailableReplicas: %v), retrying`, deploymentName, ns, deployment.Status.AvailableReplicas)
+			return false, nil
+		}
+		return true, nil
+	})
+	o.Expect(err).NotTo(o.HaveOccurred())
+}
+
+// used to wait the istio is created successfully by checking its readyReplicas
+func pollWaitIstioCreated(oc *exutil.CLI, ingressNamespace, istioName string) {
+	err := wait.Poll(3*time.Second, 300*time.Second, func() (bool, error) {
+		readyReplicas, err := oc.AsAdmin().WithoutNamespace().Run("get").Args("-n", ingressNamespace, "istio/"+istioName, `-o=jsonpath={.status.revisions.ready}`).Output()
+		if err != nil {
+			e2e.Logf("Failed to check istio %s, error: %v, retrying...", istioName, err)
+			return false, nil
+		}
+
+		if readyReplicas != "1" {
```

Comment:
May be it's worth converting `readyReplicas` to an `int` and do `readyReplicas < 1` type of condition. This way we will make the test more resilient to any changes on the number of replicas.